bundles/model/org.eclipse.smarthome.model.core/src/main/java/org/eclipse/smarthome/model/core/internal/folder/FolderObserver.java

```diff
@@ -261,23 +262,25 @@ public boolean accept(File dir, String name) {
     }
 
     @SuppressWarnings("rawtypes")
-    private static void checkFile(ModelRepository modelRepo, final File file, Kind kind) {
+    private static void checkFile(final ModelRepository modelRepo, final File file, final Kind kind) {
         if (modelRepo != null && file != null) {
             try {
                 synchronized (FolderObserver.class) {
-                    if ((kind == ENTRY_CREATE || kind == ENTRY_MODIFY) && file != null) {
+                    if ((kind == ENTRY_CREATE || kind == ENTRY_MODIFY)) {
                         if (parsers.contains(getExtension(file.getName()))) {
-                            modelRepo.addOrRefreshModel(file.getName(), FileUtils.openInputStream(file));
+                            try (FileInputStream inputStream = FileUtils.openInputStream(file)) {
+                                modelRepo.addOrRefreshModel(file.getName(), inputStream);
+                            }
                         } else {
                             ignoredFiles.add(file);
                         }
-                    } else if (kind == ENTRY_DELETE && file != null) {
+                    } else if (kind == ENTRY_DELETE) {
                         modelRepo.removeModel(file.getName());
                     }
                 }
-            } catch (IOException e) {
-                LoggerFactory.getLogger(FolderObserver.class)
-                        .warn("Cannot open file '" + file.getAbsolutePath() + "' for reading.", e);
+            } catch (Exception e) {
+                LoggerFactory.getLogger(FolderObserver.class).error(
+                        "Error handling update of file '" + file.getAbsolutePath() + "': {}.", e.getMessage(), e);
```

Comment:
I know it has not been introduced by PR, but as you touched the lines, could you fix the string concatenation?